service/controller.go

```diff
@@ -297,17 +304,23 @@ func (s *service) CreateVolume(
 		// use the default if not set in the storage class
 		rootClientEnabled = RootClientEnabledParamDefault
 	}
-	//Setting Soft Limit
+	// Setting Soft Limit
 	if _, ok := params[SoftLimitParam]; ok {
 		if params[SoftLimitParam] == "" {
 			softLimit = SoftLimitParamDefault
 		} else {
 			softLimit = params[SoftLimitParam]
 		}
 	} else {
-		// use the default if not set in the storage class
+		// use the default if not set  in the storage class
 		softLimit = SoftLimitParamDefault
 	}
+	// If value is passed in pvc than it should get precedence
```

Comment:
Pls create a new method by passing "params" and return the needed variables (after assigning values).